addons/base_import_module/models/ir_module.py

```diff
@@ -438,6 +438,11 @@ def _get_industry_categories_from_apps(self):
             return []
         except requests.exceptions.ConnectionError:
             return []
+    
+    def button_upgrade(self):
+        super().button_upgrade()
+        # revert states for imported modules since they cannot be upgraded
+        self.search([('imported', '=', True), ('state', '=', 'to upgrade')]).state = 'installed'
```

Comment:
Maybe just filter the modules before calling super?

Reply:
Why a search and not a filtered?

Reply:
> Why a search and not a filtered?

Because other modules than `self` may be marked as "to upgrade".  You want to catch those as well, I guess.

Reply:
Assume imported module Y depends on module X.  Marking module X (`self`) for upgrade automatically marks module Y for upgrade.  The point is to not upgrade module Y.  @HydrionBurst is that correct?

Reply:
Because the `button_upgrade` doesn't return those `to upgrade` module
As I described in the commit message
1. if we only upgrade official module `purchase_product_matrix`
2. the imported module `automobile` will be marked `to upgrade` after `super().button_upgrade()`

Since `'automobile' not in self.mapped('name')`, so we cannot use `filtered`